src/ucs/type/serialize.h

```diff
@@ -0,0 +1,73 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCS_SERIALIZE_H
+#define UCS_SERIALIZE_H
+
+#include <ucs/sys/compiler_def.h>
+
+/*
+ * Helper macros for serializing/deserializeing custom data
+ */
+
+/**
+ * Advance '_ptr' to the next element, and return a typed pointer to the current
+ * element.
```

Comment:
wired logic.. I would expect that `next_ptr` would return advanced ptr

Reply:
maybe rename to "ucs_serialize_advance"?
IMO it's more convenient to convert the type of current field and move to next field in same operation